src/Stack/Types/Version.hs

```diff
@@ -64,6 +65,9 @@ instance Exception VersionParseFail
 instance Show VersionParseFail where
     show (VersionParseFail bs) = "Invalid version: " ++ show bs
 
+-- | A package version or the latest.
+data UpgradeTo = Specific Version | Latest deriving (Show)
+
```

Comment:
The description is way more general than the name of the type. Can you come up with a more general name maybe?